### PR TITLE
add some common vim normal mode text commands #193

### PR DESCRIPTION
**Pull request by sodiumjoe**
movement:
b
w
e
0
$
h
l

change:
cc
cw

delete:
dd

append:
a
A

insert:
I

### CODE REVIEW

**Comment by Shougo** [comment]
No documetation.

**Comment by sodiumjoe** [comment]
OK, I'll add it.

**Comment by sodiumjoe** [comment]
added

**Comment by Shougo** [comment]
@lambdalisue Please review the changes.

File: doc/denite.txt [hunk 1]
```diff
@@ -610,8 +638,22 @@ All mode mappings (override by mode mappings).
...
 k		<denite:move_to_previous_line>
+b		<denite:move_caret_to_one_word_left>
+w		<denite:move_caret_to_next_word>
+e		<denite:move_caret_to_end_of_word>
+0		<denite:move_caret_to_head>
+$		<denite:move_caret_to_tail>
+c		<denite:change_word>
+cc	<denite:change_line>
+cw	<denite:change_word>
+dd	<denite:delete_entire_text>
```
Comment (lambdalisue) [inline doc/denite.txt]: Indent (above three lines)

File: rplugin/python3/denite/ui/action.py [hunk 2]
```diff
@@ -134,6 +136,61 @@ def _change_path(prompt, params):
...
+    prompt.text = ''
+    prompt.caret.locus = 0
+    prompt.denite.change_mode('insert')
+
+
+def _move_caret_to_next_word(prompt, params):
+    pattern = re.compile('^\S+')
+    original_text = prompt.caret.get_forward_text()
+    substituted_text = pattern.sub('', original_text)
+    prompt.caret.locus += 2 + len(original_text) - len(substituted_text)
```
Comment (lambdalisue) [inline rplugin/python3/denite/ui/action.py]: `2` does not make sense to me. It should be a variable instead of a magic number.

Comment (sodiumjoe) [inline rplugin/python3/denite/ui/action.py]: I updated the regex to `^\S+\s+\S` so the magic number is no longer necessary (and it handles multiple white space character before the next word)

**Comment by Shougo** [comment]
Merged.

**Pull request merged**